diff-lib.c

```diff
@@ -197,6 +199,9 @@ int run_diff_files(struct rev_info *revs, unsigned int option)
 		if (ce_uptodate(ce) || ce_skip_worktree(ce))
 			continue;
 
+		if (ce->ce_flags & CE_FSMONITOR_VALID)
+			continue;
```

Comment:
Correct me if I'm wrong, but doesn't `CE_FSMONITOR_VALID` mean that FSMonitor claims that the metadata hasn't changed since the previous refresh? Note that if my understanding is correct, this does _not_ imply that the file is unchanged. It just means that the recorded metadata (such as `mtime`) matches what's on disk.

Reply:
This is specifically to avoid the call to `check_removed` which does an lstat - so I believe that means that CE_FSMONITOR_VALID is what we want.

Leads me to wonder if we can move this call inside check_removed